OmegaExamples/src/probmods/generative-models.jl

```diff
@@ -0,0 +1,612 @@
+"""
+
+# ╔═╡ 0ce18f90-f2ef-490a-a7c4-40a6ccfa214a
+md"""
+A basic object in Omega is a random variable class, from which you can generate random variables. For instance, a random variable that is of Bernoulli distribution is generated by -
+"""
+
+# ╔═╡ 354e3b5a-4d1e-43ee-b1bb-e8620fba49e7
+a = 0 ~ Bernoulli()
+
+# ╔═╡ 4c915901-e987-4552-b5f7-bc25ee6edeec
+md"Here, `Bernoulli()` is a random class and to get a member (a random variable, which can be sampled from) of the class we use `~`. So, `0 ~ Bernoulli()` is the `0`th member of Bernoulli class."
+
+# ╔═╡ 7d3c8448-090d-46b4-a6b6-ba20ff617be8
+md"A class is analogous to a [*plate*](https://en.wikipedia.org/wiki/Plate_notation) in Bayesian networks."
+
+# ╔═╡ 9dca3d31-8ace-4c64-996c-d9d9e87e6ee4
+md"### Sampling "
+
+# ╔═╡ cc0c0d70-5197-4b2f-b32d-6ac52b8daf27
+md"`randsample` is used to sample from a random variable:"
+
+# ╔═╡ d607d2e0-84a5-4dfd-a281-f5c64cfe1e2d
+randsample(a)
+
+# ╔═╡ 5a10bc0d-d676-4941-a1d5-41790ef255ec
+md"Run this program a few times. You will get back a different sample on each execution."
+
+# ╔═╡ 02ca238e-5a17-42dc-92a1-be975db0e7cd
+md"To understand how `randsample` works, a key concept to know is the Ω object in Omega.jl."
+
+# ╔═╡ 4ea27371-2e73-4a69-887c-b9f519fbf583
+md"""
+There are a few perspectives on what Ω is:
+
+1. The sample space in probability theory (often denoted $\Omega$)
+2. Space of exogenous variables in as causal graphical modelling nomenclature (often denoted $U$)
+3. The random number generator `rng::Random.AbstractRNG`.  Implementation wise, Omega.jl tries to be compatible with any model written as a function `f(rng::AbstractRNG)`
+"""
+
+# ╔═╡ d70b97b5-af55-46f9-904e-309713603155
+md"""
+Conceptually, sampling from a random variable is done in the following two distinct steps:
+1. Construct a random $ω : Ω$, where $Ω$ can be though of as the space of exogenous variables as in causal graphical modelling nomenclature or simply the random number generator, `rng::Random.AbstractRNG`.
+2. Apply $ω$ to the random variable : $A(ω)$
+"""
+
+# ╔═╡ 0360ccef-ccf4-4284-80a9-ee5ed40becc4
+md"### Primitive and Composite Classes in Omega"
+
+# ╔═╡ 9998d466-eb74-476f-a768-d378bbe83f1b
+md"""##### Primitive Classes
+Omega comes with a set of built-in primitive random variable classes, such as `StdNormal` and `StdUniform`.  There are parameterless infinite sets of random variables.
+"""
+
+# ╔═╡ a0d64a28-e4f9-4003-b69d-7b27bd757c24
+As = StdUniform{Float64}()
+
+# ╔═╡ 84acb184-1c92-4761-a44e-bd1503f8255d
+rv = 1 ~ As
+
+# ╔═╡ 4e80ce71-6757-4908-8cd0-e382b11b70c3
+randsample(rv)
+
+# ╔═╡ 5fd22c50-4c81-41b2-a959-fe3dec00a6fd
+md"""
+##### Composite Class
+
+A class in Omega is actually just function of the form f(id, ω). Of course, you can specify your own classes simply by constructing a function.
+"""
+
+# ╔═╡ 8ac4cad4-eb4c-4e33-815c-e365668b3810
+μ = 2 ~ StdNormal{Float64}()
+
+# ╔═╡ 5fcf6c97-6f81-4be2-b27a-25b2abcabca1
+x(id, ω) = (id ~ Normal(μ(ω), 1))(ω)
```

Comment:
Ah, so μ, an RV, is callable with ω as an argument.  This is the first time I'm seeing this pattern in this tutorial, as is the composite non-standard Normal you construct.  Is it true that all primitive classes don't take ω as an argument and all composite classes do?  If so, say so.

The use of ω as an exposed thread that ties together a sequence of expressions is a fundamentally unique element Omega.  I suggest at least explicitly highlighting that fact here.  Explaining it here would be ideal, but perhaps it is explained elsewhere in the docs.  If so, please add a link to that doc.